### PR TITLE
Adjust docs to not include submodules anymore

### PR DESCRIPTION
**Pull request by ItsShadowCone**
Adjusting docs accordingly to this pr: lirios/lirios#48

### CODE REVIEW

File: docs/contributor-guide/index.md [hunk 1]
```diff
@@ -167,17 +166,3 @@ Make sure all your changes are either committed or stashed and run:
...
 repo sync
 ```
-
-Make sure the branch is checked out and all submodules are updated:
-
-```sh
-repo forall -c 'git checkout $REPO_RREV'
-repo forall -c 'git submodule update --init --recursive'
-```
-
```
Comment (plfiorini) [inline docs/contributor-guide/index.md:177]: We should keep the instructions below.
The commit template helps writing good commit messages.

**Review by plfiorini — CHANGES_REQUESTED** [review]

File: docs/contributor-guide/index.md (hunk 1 above)
Comment (ItsShadowCone) [reply docs/contributor-guide/index.md:177]: This section is about updating the repo source. it is only necessary to run "repo sync", as one had set the commit template earlier (or did i miss that repo resets that?)

Comment (plfiorini) [reply docs/contributor-guide/index.md:177]: a new repository might be added to the manifest file in between `repo sync`s so developers should do the same steps for the new repositories this is why the same commands are repeated.

**Comment by ItsShadowCone** [comment]
Please, go ahead and squash merge :)

**Review by plfiorini — APPROVED** [review]

**Pull request merged**